src/code/sys_cfb.c

```diff
@@ -1,20 +1,128 @@
+void* sCfbLoRes0;
+u16 (*gZBufferLoRes)[SCREEN_WIDTH * SCREEN_HEIGHT];
+u16 (*gWorkBufferLoRes)[SCREEN_WIDTH * SCREEN_HEIGHT];
+u64 (*gGfxSPTaskOutputBufferLoRes)[0x3000];
+void* gGfxSPTaskOutputBufferSizeLoRes; // Actually points to the end of the task buffer
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/func_801789D4.s")
+void* sCfbHiRes1;
+void* sCfbHiRes0;
+u16 (*gZBufferHiRes)[SCREEN_WIDTH_NOTEBOOK * SCREEN_HEIGHT_NOTEBOOK];
+u16 (*gWorkBufferHiRes)[SCREEN_WIDTH_NOTEBOOK * SCREEN_HEIGHT_NOTEBOOK];
+u64 (*gGfxSPTaskOutputBufferHiRes)[0x3000];
+void* gGfxSPTaskOutputBufferSizeHiRes; // Actually points to the end of the task buffer
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/SysCfb_GetFbPtr.s")
+s16 gCfbWidth;
+s16 gCfbHeight;
+s16 gCfbLeftAdjust;
+s16 gCfbUpperAdjust;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/SysCfb_GetZBuffer.s")
+u8 gSysCfbHiResEnabled;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/func_80178A24.s")
+void SysCfb_SetLoResMode(void) {
+    gFramebuffers[1] = sCfbLoRes1;
+    gFramebuffers[0] = sCfbLoRes0;
+    gZBufferPtr = *gZBufferLoRes;
+    gWorkBuffer = gWorkBufferLoRes;
+    gGfxSPTaskOutputBufferPtr = *gGfxSPTaskOutputBufferLoRes;
+    gGfxSPTaskOutputBufferSize = gGfxSPTaskOutputBufferSizeLoRes;
+    gCfbWidth = SCREEN_WIDTH;
+    gCfbHeight = SCREEN_HEIGHT;
+    gCfbLeftAdjust = 0;
+    gCfbUpperAdjust = 0;
+    gSysCfbHiResEnabled = false;
+    gScreenWidth = gCfbWidth;
+    gScreenHeight = gCfbHeight;
+    gActiveViMode = &osViModeNtscLan1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/func_80178A34.s")
+void SysCfb_SetHiResMode(void) {
+    gFramebuffers[1] = sCfbHiRes1;
+    gFramebuffers[0] = sCfbHiRes0;
+    gZBufferPtr = *gZBufferHiRes;
+    gWorkBuffer = gWorkBufferHiRes;
+    gGfxSPTaskOutputBufferPtr = *gGfxSPTaskOutputBufferHiRes;
+    gGfxSPTaskOutputBufferSize = gGfxSPTaskOutputBufferSizeHiRes;
+    if (1) {}
+    gCfbWidth = SCREEN_WIDTH_NOTEBOOK;
+    gCfbHeight = SCREEN_HEIGHT_NOTEBOOK;
+    gCfbLeftAdjust = 30;
+    gCfbUpperAdjust = 10;
+    gScreenWidth = gCfbWidth;
+    gScreenHeight = gCfbHeight;
+    if ((gCfbWidth == SCREEN_WIDTH_HIRES) && (gCfbHeight == SCREEN_HEIGHT_HIRES)) {
+        gActiveViMode = &osViModeNtscHpf1;
+    } else {
+        s32 rightAdjust;
+        s32 lowerAdjust;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/sys_cfb/func_80178A94.s")
+    //! FAKE:
+    l1:
+        rightAdjust = gCfbWidth - 610;
+        lowerAdjust = gCfbHeight - 470;
+        ViMode_Configure(&sNotebookViMode, -1, osTvType, 0, 1, 0, 1, gCfbWidth, gCfbHeight, 30, rightAdjust, 10,
+                         lowerAdjust);
+        gActiveViMode = &sNotebookViMode;
+    }
+    gSysCfbHiResEnabled = true;
+}
+
+//! FAKE:
+extern u16 gFramebuffer1_[SCREEN_WIDTH_NOTEBOOK][SCREEN_HEIGHT_NOTEBOOK];
+void SysCfb_Init(void) {
+    sCfbLoRes1 = gFramebuffer1;
+    sCfbLoRes0 = gFramebuffer0;
+    sCfbHiRes1 = gFramebuffer1_;
+    sCfbHiRes0 = D_80780000;
```

Comment:
Do you know what the 0 and 1 are?
If I had to guess it seems like 0 points to the start of the buffer and 1 points to the end of the buffer. Maybe We could  improve symbols here based on that assumption?

Reply:
I assume you're asking about gFramebuffer0 and gFramebuffer1. Best way to move forward on this is to look at how everything is getting mapped to memory

Quick address map:

* `80000500` is a buffer for sCfbLoRes1 and sCfbHiRes1
* `80025D00` is a buffer for the skybox data
* `80780000` is a buffer for sCfbHiRes0 and gPictoPhotoI8
* `80784600` is a buffer for various effects (lens, maybe the screen shrink effect)
* `807DA800` maps to sCfbLoRes0

The lores color framebuffer is 320 x 240 pixels, 0x25800 bytes
The highres color framebuffer is 576 x 454 pixels, 0x7FB00 bytes
The pictobox photo texture buffer is 160 x 112 pixels, 0x4600 bytes
The buffer following gPictoPhotoI8 is defined to be 0x56200 bytes

If you look at the big picture, essentially you have a 0x7FB00 block of memory at `80000500` to `80080000` and a 0x80000 block of memory at `80780000` to `80800000`. So the names `gFramebuffer0` and `gFramebuffer1` are aren't correct for universal usage like it is atm.

Reply:
No, I was asking about `sCfbLoRes1` and `sCfbLoRes0`

Reply:
Also, are you saying `D_80780000` should be a `0x7FB00` long buffer? Because if that's the case then that buffer would overlap with `gFramebuffer0`, which is at `0x807DA800`
To avoid overlap that buffer must be at most `0x5A800`. This would allow for a buffer which is at most 497×372 pixels
What exactly makes you think this framebuffer should be 576 x 454 pixels?

Reply:
Cfb is the abbreviation for color framebuffer, so these are the two framebuffers used during regular play; the 0 and 1 just indicates what buffer index they are assigned to in gFramebuffers.

Low resolution and high resolution modes are mutually exclusive, so to save ram the high res framebuffers overlap the other buffers.

> Also, are you saying D_80780000 should be a 0x7FB00 long buffer?

Not quite, I'm saying it's an 0x80000 long buffer, since when you're in the game normally you have the 0x4600, 0x56200, and 0x25800 buffers reserved there.

> What exactly makes you think this framebuffer should be 576 x 454 pixels?

`SysCfb_SetHiResMode` configures high resolution mode; within it it sets gCfbWidth, gCfbHeight to 576 by 454

Reply:
Then I propose to do this then:
```suggestion
    sCfbLoRes1 = gFramebuffer1;
    sCfbLoRes0 = gFramebuffer0;
    sCfbHiRes1 = gFramebufferHiRes1;
    sCfbHiRes0 = gFramebufferHiRes0;
```
And add those two extra symbols to undefined_syms.txt
```
gFramebufferHiRes1 = 0x80000500;
gFramebufferHiRes0 = 0x80780000;
```